### PR TITLE
Unit test for CustomRequestScheduleManager

### PR DESCRIPTION
**Pull request by lkomali**
This PR
1. Creates Unit test for CustomRequestScheduleManager
2. Corrects the schedule used while creating workers (used schedule instead of scaled_schedule)
3. Fixes segmentation fault that is occurred when schedule.size() < 4.
4. Modifies the param variable passed, earlier was passing a shared_ptr to params, but now modified to passing params object to make the unit test work.
5. Update GenerateSchedule function to only take request_rate, as schedule is an attribute of the class and need not be passed.

### CODE REVIEW

File: src/custom_request_schedule_manager.h
Comment (matthewkotila) [inline src/custom_request_schedule_manager.h:58]: I'm a fan of passing references over smart pointers, but why was this change needed?

Comment (lkomali) [reply src/custom_request_schedule_manager.h:58]: For the unit test, I don't remember quite correctly, but the unit test couldn't access the pointer.
I will take a look later to refresh my memory and correctly define what was going wrong.

File: src/test_custom_request_schedule_manager.cc
Comment (dyastremsky) [inline src/test_custom_request_schedule_manager.cc:75]: Are the unit tests only testing that the correct number of timestamps are provided? Do the unit tests also need to make sure that the timestamps are the ones expected?

Comment (lkomali) [reply src/test_custom_request_schedule_manager.cc:75]: As far as I understand, it's checking if the timestamps calculated in the test match the timestamps the workers are scheduled at. 
I have taken reference to this [check](https://github.com/triton-inference-server/perf_analyzer/blob/main/src/test_request_rate_manager.cc#L149) in **test_request_rate_manager.cc**
The .count() gives duration as an integer. 
Essentially, we are checking that the requests are scheduled at expected timestamps.

File: src/test_custom_request_schedule_manager.cc
Comment (lkomali) [reply src/test_custom_request_schedule_manager.cc:37]: I will remove this line before merging.

File: src/test_custom_request_schedule_manager.cc
Comment (dyastremsky) [reply src/test_custom_request_schedule_manager.cc:75]: I see, thanks for explaining. I believe you can check if the timestamps (std::chrono::nanoseconds) are equal directly and I don't believe the .count() adds anything to this check, but I'm okay with us leaving it if that's preferred. It's not worth slowing down this getting in.

File: src/CMakeLists.txt
Comment (matthewkotila) [inline src/CMakeLists.txt:1]: PR LGTM, just waiting on other reviewers to suggest approval.

Comment (dyastremsky) [reply src/CMakeLists.txt:1]: Testing portion of the PR LGTM as well.

**Review by matthewkotila — APPROVED** [review]

**Pull request merged**
